codegenerator/model/api.go

```diff
@@ -193,15 +186,18 @@ func New(credentials *tcclient.Credentials) *` + api.apiDef.Name + ` {
 //  TASKCLUSTER_CLIENT_ID
 //  TASKCLUSTER_ACCESS_TOKEN
 //  TASKCLUSTER_CERTIFICATE
+//  TASKCLUSTER_ROOT_URL
+//
+// If environment variable TASKCLUSTER_ROOT_URL is empty string or not set,
+// https://taskcluster.net will be assumed.
```

Comment:
We have not set a default value for rootUrl in the JS client.  I worry that this is going to hide places where we missed setting rootUrl, letting them appear when we shut down `taskcluster.net`.  Admittedly, there are [fewer of those to worry about](https://godoc.org/github.com/taskcluster/taskcluster-client-go?importers) than for JS or Python, since basically only generic-worker uses the go client.  Still, I'd like such errors to surface as quickly and easily as possible.  For example, that made porting services to the new version of the JS client *much* easier, by just running the tests until they stopped failing with "rootUrl must be specified" :)

That new requirement is the reason for the major version change on the JS and Python libraries.  I don't know if it's worth doing the same for this library (it requires a `v2/` directory to make a breaking change in Go, right?), or given the limited usage of the library we can slide the change in in the same repository.

Reply:
It is funny, I had it in mind to go back and change this, it looks like I must have forgotten to - but I came to the same reasoning, so thanks for reminding me! Indeed, we also don't want external organisations to run their client, forget to export `TASKCLUSTER_ROOT_URL` in their environment (for example) and then suddenly (and surprisingly) connect to our production cluster. In the worst case they might call an unauthenticated endpoint (such as listing clients) and actually get back real data, rather than fail out with an error. That indeed would be undesirable! So, I agree 100%, and am also grateful you took the time to look through the code to see what it was doing. :-)